code/modules/power/cell.dm

```diff
@@ -118,9 +118,7 @@
 		rigged = 1 //broken batterys are dangerous
 
 /obj/item/weapon/cell/emp_act(severity)
-	charge -= 1000 / severity
-	if (charge < 0)
-		charge = 0
+	charge = max(charge - maxcharge/severity, 0)
```

Comment:
I feel like everything about this line is wrong. Charge-maxcharge will always give a negative number, and high severities are weaker (3 is edge, 1 is middle) iirc.

So basically since this will always be a negative number or zero, it will max to 0.